cmd/publish.go

```diff
@@ -0,0 +1,155 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/avorty/spito/cmd/cmdApi"
+	"github.com/avorty/spito/internal/checker"
+	"github.com/avorty/spito/pkg/shared"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	publishRulesetRoute = "ruleset/publish"
+)
+
+type RuleForRequest struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Path        string `json:"path"`
+	Unsafe      bool   `json:"unsafe"`
+}
+
+type PublishRequestBody struct {
+	Url   string           `json:"url"`
+	Rules []RuleForRequest `json:"rules"`
+}
+
+var publishCommand = &cobra.Command{
+	Use:   "publish [-l|--local] [ruleset_path]",
+	Short: "Publish a ruleset to spito store",
+	Args:  cobra.MaximumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		var rulesetPath string
+		var err error
+
+		if len(args) > 0 && args[0] != "" {
+			rulesetPath = args[0]
+		} else {
+			rulesetPath, err = os.Getwd()
+			handleError(err)
+		}
+
+		configFileContents, err := os.ReadFile(filepath.Join(rulesetPath, checker.ConfigFilename))
+		if os.IsNotExist(err) {
+			printErrorAndExit(errors.New("please point this command to an actual spito ruleset"))
+		} else {
+			handleError(err)
+		}
+
+		var configFileValues ConfigFileLayout
+		err = yaml.Unmarshal(configFileContents, &configFileValues)
+		handleError(err)
+
+		requestBody := PublishRequestBody{
+			Url: configFileValues.Repo_url,
+		}
+		for ruleName, rule := range configFileValues.Rules {
```

Comment:
separate this entire logic responsible for getting RuleConfig into other function (I think this function should be putted into `checker` package, instead of `cmd`)